### PR TITLE
Replace RAW SQL by new querying syntax

### PR DESCRIPTION
**Pull request by cedric-anne**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

### CODE REVIEW

File: inc/contract.class.php [hunk 1]
```diff
@@ -1202,42 +1207,113 @@ static function cronContract($task = null) {
...
-                             AND `glpi_contracts`.`duration` <> '0'
-                             AND DATEDIFF(ADDDATE(`glpi_contracts`.`begin_date`,
-                                                  INTERVAL (`glpi_contracts`.`duration`) MONTH),
-                                          CURDATE()) < '$before'
-                             AND `glpi_alerts`.`date` IS NULL
-                             AND `glpi_contracts`.`entities_id` = '".$entity."'";
+
+         $query_notice = [
+            'SELECT'    => [
+               'glpi_contracts.*',
```
Comment (trasher) [inline inc/contract.class.php]: useless?

File: inc/cartridgeitem.class.php [hunk 2]
```diff
@@ -458,22 +458,39 @@ static function cronCartridge($task = null) {
...
+                     ]
+                  ]
+               ],
+               'WHERE'     => [
+                  'glpi_cartridgeitems.is_deleted'      => 0,
+                  'glpi_cartridgeitems.alarm_threshold' => ['>=', 0],
+                  'glpi_cartridgeitems.entities_id'     => $entity,
+                  'OR'                                  => [
+                     ['glpi_alerts.date' => null],
+                     ['glpi_alerts.date' => ['<', new QueryExpression('CURRENT_TIMESTAMP() - INTERVAL ' . $repeat . ' second')]],
```
[the remark this answers is not in the document]
Comment (cedric-anne) [reply inc/cartridgeitem.class.php]: Yes, to fix a bug like in https://github.com/glpi-project/glpi/commit/ef483e142cda912e6cb7576f454434fc55ec1daa#diff-5c560cccdd1a95b3ea803ec080824ba8.

File: inc/contract.class.php (hunk 1 above)
Comment (cedric-anne) [reply inc/contract.class.php]: Not sure, request also includes alerts table and I do not know how ambiguous field names are handled by SQL and/or PHP.

Comment (trasher) [reply inc/contract.class.php]: OK for me then

File: inc/cartridgeitem.class.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (trasher) [reply inc/cartridgeitem.class.php]: Hum... OK, was not clear at all ;)

File: inc/computerantivirus.class.php [hunk 3]
```diff
@@ -335,73 +337,74 @@ static function showForComputer(Computer $comp, $withtemplate = 0) {
...
       echo "<div class='spaced center'>";
 
-      $query = [
-         'FROM'  => 'glpi_computerantiviruses',
-         'WHERE' => [
-            'computers_id' => $ID,
-            'is_deleted'   => 0,
-         ],
-      ];
-      if ($result = $DB->request($query)) {
```
[the remark this answers is not in the document]
Comment (trasher) [reply inc/computerantivirus.class.php]: ok

**Pull request merged**